staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -283,6 +286,12 @@ func (flags *ApplyFlags) ToOptions(f cmdutil.Factory, cmd *cobra.Command, baseNa
 	}
 
 	openAPISchema, _ := f.OpenAPISchema()
+	var openAPIV3Root openapi3.Root
+	openAPIV3Client, err := f.OpenAPIV3Client()
+	if err == nil {
+		cachedOpenAPIV3Client := cachedopenapi.NewClient(openAPIV3Client)
```

Comment:
Probably not for now, but we should probably have the factory return a cachedopenapi.Client, that's already what we do for v2. WDYT @seans3

Reply:
It's a fairly simple change, I changed the factory to use a cached openapi v3 client.

Reply:
Sounds good (on the `cachedopenapi.Client`).